ocs_ci/deployment/helpers/prechecks.py

```diff
@@ -0,0 +1,162 @@
+    Should be inherited by specific platform classes
+    """
+    def __init__(self):
+        """
+        Initialize required variables
+        """
+        self.cluster_path = config.ENV_DATA['cluster_path']
+        self.platform = config.ENV_DATA['platform']
+        self.deployment_type = config.ENV_DATA['deployment_type']
+
+    def storage_check(self):
+        raise NotImplementedError(
+            "storage check functionality is not implemented"
+        )
+
+    def memory_check(self):
+        raise NotImplementedError(
+            "memory check functionality is not implemented"
+        )
+
+    def cpu_check(self):
+        raise NotImplementedError(
+            "CPU check functionality is not implemented"
+        )
+
+    def network_check(self):
+        raise NotImplementedError(
+            "Network check functionality is not implemented"
+        )
+
+
+class VSPHERE_PRECHECKS(PRECHECKS):
+    """
+    Pre-checks for vSphere platform
+    """
+    def __init__(self):
+        """
+        Initialize required variables
+        """
+        super(VSPHERE_PRECHECKS, self).__init__()
+        self.server = config.ENV_DATA['vsphere_server']
+        self.user = config.ENV_DATA['vsphere_user']
+        self.password = config.ENV_DATA['vsphere_password']
+        self.datacenter = config.ENV_DATA['vsphere_datacenter']
+        self.datastore = config.ENV_DATA['vsphere_datastore']
+        self.vsphere = VSPHEREUtil(self.server, self.user, self.password)
+
+    def storage_check(self):
+        """
+        Checks for storage capacity in the datastore
+
+        Raises:
+            StorageNotSufficientException: In case if there is no sufficient
+                storage in Datastore.
+
+        """
+        logger.debug(f"Checking for datastore {self.datastore} free capacity")
+        datastore_free_capacity = self.vsphere.get_datastore_free_capacity(
+            self.datastore,
+            self.datacenter
+        )
+        if datastore_free_capacity < MIN_STORAGE_FOR_DATASTORE:
+            raise StorageNotSufficientException
+
+    def memory_check(self):
+        """
+        Memory checks
+        """
+        raise NotImplementedError(
+            "memory check functionality is not implemented"
+        )
+
+    def cpu_check(self):
+        """
+        CPU checks
+        """
+        raise NotImplementedError(
+            "CPU check functionality is not implemented"
+        )
+
+    def network_check(self):
+        """
+        Network related checks
+        """
+        raise NotImplementedError(
+            "Network check functionality is not implemented"
+        )
+
+    def get_all_checks(self):
+        """
+        Aggregate all the checks needed for vSphere platform
+        """
+        self.storage_check()
+
+
+class BM_PRECHECKS(PRECHECKS):
+    """
+    pre-checks for Bare Metal platform
+    """
+    def __init__(self):
+        super(VSPHERE_PRECHECKS, self).__init__()
```

Comment:
This should be `BM_PRECHECKS` (or `BareMetalPreChecks` if we change to CapWords convention).

Reply:
Done